src/ecmascript/spidermonkey/localstorage-db.c

```diff
@@ -0,0 +1,207 @@
+/* The SpiderMonkey localstorage database helper implementation. */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <sqlite3.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "elinks.h"
+#include "src/ecmascript/ecmascript.h"
+
+extern const int
+db_prepare_structure(char *db_name)
+{
+	sqlite3_stmt *stmt;
+	sqlite3 *db;
+
+	int rc;
+	char *sqlbuffer;
+
+	rc = sqlite3_open(db_name, &db);
+	if (rc)
+	{
+		//DBG("Error opening localStorage database.");
+		rc=sqlite3_close(db);
+		return(-1);
+	}
+	sqlite3_busy_timeout(db, 5000);
+	sqlbuffer=stracpy("CREATE TABLE storage (key TEXT, value TEXT);");
+	rc=sqlite3_prepare_v2(db, sqlbuffer, strlen(sqlbuffer) + 1, &stmt, NULL);
+	rc=sqlite3_step(stmt);
+	rc=sqlite3_finalize(stmt);
+	rc=sqlite3_close(db);
+	return(0);
+}
+
+extern const int
+db_delete_from(char *db_name, char *key)
+{
+
+	sqlite3_stmt *stmt;
+	sqlite3 *db;
+
+	int rc;
+	char *sqlbuffer;
+	int affected_rows = 0;
+
+	rc = sqlite3_open(db_name, &db);
+	if (rc)
+	{
+		//DBG("Error opening localStorage database.");
+		rc=sqlite3_close(db);
+		return(-1);
+	}
+	sqlite3_busy_timeout(db, 5000);
+	sqlbuffer=stracpy("DELETE FROM storage WHERE key = ?;");
+	rc=sqlite3_prepare_v2(db, sqlbuffer, strlen(sqlbuffer) + 1, &stmt, NULL);
+	rc=sqlite3_bind_text(stmt, 1, key, strlen(key), SQLITE_STATIC);
+	rc=sqlite3_step(stmt);
+	rc=sqlite3_finalize(stmt);
+	affected_rows=sqlite3_changes(db);
+	rc=sqlite3_close(db);
+	return(affected_rows);
+
+}
+
+
+extern const int
+db_insert_into(char *db_name, char *key, char *value)
+{
+	sqlite3_stmt *stmt;
+	sqlite3 *db;
+
+	int rc;
+	char *sqlbuffer;
+	int affected_rows = 0;
+
+	rc = sqlite3_open(db_name, &db);
+	if (rc) {
+		//DBG("Error opening localStorage database.");
+		rc=sqlite3_close(db);
+		return("");
+	}
+	sqlite3_busy_timeout(db, 5000);
+	sqlbuffer=stracpy("INSERT INTO storage (value,key) VALUES (?,?);");
+	rc=sqlite3_prepare_v2(db, sqlbuffer, strlen(sqlbuffer) + 1, &stmt, NULL);
+	rc=sqlite3_bind_text(stmt, 1, value, strlen(value), SQLITE_STATIC);
+	rc=sqlite3_bind_text(stmt, 2, key, strlen(key), SQLITE_STATIC);
+	rc=sqlite3_step(stmt);
+	rc=sqlite3_finalize(stmt);
+	affected_rows=sqlite3_changes(db);
+	rc=sqlite3_close(db);
+	return(affected_rows);
+
+}
+
+extern const int
+db_update_set(char *db_name, char *key, char *value)
+{
+
+	sqlite3_stmt *stmt;
+	sqlite3 *db;
+
+	int rc;
+	char sqlbuffer[8192];
+	char result[8192];
+	int affected_rows = 0;
+
+	rc = sqlite3_open(db_name, &db);
+	if (rc) {
+		//DBG("Error opening localStorage database.");
+		rc=sqlite3_close(db);
+		return("");
+	}
+	sqlite3_busy_timeout(db, 5000);
+	snprintf(sqlbuffer, 256, "UPDATE storage SET value = ? where key = ?;");
```

Comment:
Here too, you do not have call snprintf. Just pass query as string directly to sqlite3_prepare_v2(db,  "UPDATE storage SET value = ? WHERE key = ?;", ...)

Reply:
O.K. agreed, Fixed.